### PR TITLE
feat(transforms): add 8 robotics-relevant image augmentations

### PR DESCRIPTION
**Pull request by Liyux3**
## Summary / Motivation

LeRobot's default image transforms are photometric or geometric. However, real-robot deployment consistently surfaces failure modes that these don't cover: sensor noise in low-light wrist cameras, motion blur during fast contact manipulation, compression artifacts from streamed video, uneven workspace lighting, cast shadows from humans and objects, and partial occlusion from cables and tools.

[RAD](https://arxiv.org/abs/2004.14990) showed that augmentations as simple as random crop improve sample efficiency by 5.75x in visual RL. [DrQ](https://arxiv.org/abs/2004.13649) demonstrated that a single well-chosen augmentation matches methods requiring auxiliary losses or contrastive pretraining. [GreenAug](https://arxiv.org/abs/2407.07868) found lighting and shadow variation to be the dominant generalization bottleneck in real-robot manipulation across 850 demos and 8200 eval episodes. Yet every major robotics framework, including Diffusion Policy, ACT, pi0, Octo, and OpenVLA, still ships at most color jitter and random crop.

This PR adds 8 transforms that address the most common gaps: `GaussianNoise`, `MotionBlur`, `JPEGCompression`, `GaussianPatchBrightness`, `RandomShadow`, `CoarseDropout`, `GammaCorrection`, and [`PlanckianJitter`](https://arxiv.org/abs/2202.07993). All are pure PyTorch with no external dependencies and integrate with `ImageTransformConfig` requiring zero code changes from users.

Validated on a real 3-camera manipulation setup (793 episodes, flow matching policy), where comprehensive augmentation yields measurably better generalization compared to defaults alone.

<details>
<summary>Visual effect of each transform</summary>

aug_showcase

Sample frame from [NVIDIA PhysicalAI-Robotics-Manipulation-SingleArm](https://huggingface.co/datasets/nvidia/PhysicalAI-Robotics-Manipulation-SingleArm) (CC-BY-4.0).

</details>

## Related issues

None open. This addresses a gap in the current transform set.

## What changed

- Added 8 new `Transform` subclasses to `src/lerobot/transforms/transforms.py`, all following the existing `make_params`/`transform` pattern
- Introduced a `_CUSTOM_TRANSFORMS` registry dict that replaces the hardcoded `SharpnessJitter` check in `make_transform_from_config`, making future additions trivial
- Updated `src/lerobot/transforms/__init__.py` to export the new classes
- Added 41 tests to `tests/datasets/test_image_transforms.py` covering shape preservation, output range, dtype, non-float passthrough, and config integration for each transform

Existing configs and `SharpnessJitter` continue to work unchanged.

## How was this tested (or how to run locally)

```bash
pytest tests/datasets/test_image_transforms.py -v
```

Each transform is tested for shape preservation, output clamped to [0, 1], float dtype output, non-float passthrough (uint8 unchanged), and correct instantiation via `make_transform_from_config`.

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Reviewed [#3882](https://github.com/huggingface/lerobot/pull/3882)

## Reviewer notes

- The registry change is backward-compatible. Existing configs and `SharpnessJitter` continue to work unchanged.
- `RandomShadow` and `GammaCorrection` use symmetric sampling (equal probability of brightening and darkening) to avoid shifting BatchNorm running statistics during training.
- `MotionBlur` uses directional `avg_pool2d` instead of constructing a convolution kernel, which is faster (~1ms vs ~20ms).
- These 8 are a subset of a larger [44-method pipeline](https://github.com/Liyux3/robo-augment) we use internally. We selected the most broadly applicable ones for upstream contribution.

### CODE REVIEW

**Comment by imstevenpmwork** [comment]
Hello @Liyux3 

Thanks for contributing! This is actually very cool. I will be doing some experiments/changes in this branch: https://github.com/huggingface/lerobot/pull/4210
Would be nice if you can take a look at let me know what you think

Kind regards!

**Comment by imstevenpmwork** [comment]
Superseded by https://github.com/huggingface/lerobot/pull/4210
